projects/packages/masterbar/src/masterbar/class-masterbar.php

```diff
@@ -0,0 +1,1570 @@
+		}
+
+		return $redirect_to;
+	}
+
+	/**
+	 * Adds CSS classes to admin body tag.
+	 *
+	 * @since jetpack-5.1
+	 *
+	 * @param string $admin_body_classes CSS classes that will be added.
+	 *
+	 * @return string
+	 */
+	public function admin_body_class( $admin_body_classes ) {
+
+		$classes = array( 'jetpack-masterbar', trim( $admin_body_classes ) );
+
+		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
+			$classes[] = 'wpcom-admin-interface';
+		}
+
+		return implode( ' ', $classes );
+	}
+
+	/**
+	 * Remove the default Admin Bar CSS.
+	 */
+	public function remove_core_styles() {
+		/*
+		 * Notifications need the admin bar styles,
+		 * so let's not remove them when the module is active.
+		 * Also, don't remove the styles if the user has opted to use wp-admin.
+		 */
+		if ( ! ( new Modules() )->is_active( 'notes' ) && get_option( 'wpcom_admin_interface' ) !== 'wp-admin' ) {
+			wp_dequeue_style( 'admin-bar' );
+		}
+	}
+
+	/**
+	 * Enqueue our own CSS and JS to display our custom admin bar.
+	 */
+	public function add_styles_and_scripts() {
+		$assets_base_path = '../../dist/masterbar/';
+
+		// WoA sites: If wpcom_admin_interface is set to wp-admin, load the wp-admin styles.
+		// These include only styles to enable the "My Sites" and "Reader" links that will be added.
+		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
+			$css_file = $this->is_rtl ? 'masterbar-wp-admin-rtl.css' : 'masterbar-wp-admin.css';
+			wp_enqueue_style( 'a8c-wpcom-masterbar-overrides', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/masterbar-overrides/' . $css_file ), array(), Main::PACKAGE_VERSION );
+			return;
+		}
+
+		if ( $this->is_rtl ) {
+			wp_enqueue_style( 'a8c-wpcom-masterbar-rtl', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/rtl/wpcom-admin-bar-rtl.css' ), array(), Main::PACKAGE_VERSION );
+			wp_enqueue_style( 'a8c-wpcom-masterbar-overrides-rtl', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/masterbar-overrides/rtl/masterbar-rtl.css' ), array(), Main::PACKAGE_VERSION );
+		} else {
+			wp_enqueue_style( 'a8c-wpcom-masterbar', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/wpcom-admin-bar.css' ), array(), Main::PACKAGE_VERSION );
+			wp_enqueue_style( 'a8c-wpcom-masterbar-overrides', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/masterbar-overrides/masterbar.css' ), array(), Main::PACKAGE_VERSION );
+		}
+
+		// Local overrides.
+		Assets::register_script(
+			'a8c_wpcom_css_override',
+			$assets_base_path . 'overrides.js',
+			__FILE__,
+			array(
+				'enqueue'  => true,
+				'css_path' => $assets_base_path . 'overrides.css',
+			)
+		);
+
+		if ( ! ( new Modules() )->is_active( 'notes' ) ) {
+			// Masterbar is relying on some icons from noticons.css.
+			wp_enqueue_style( 'noticons', $this->wpcom_static_url( '/i/noticons/noticons.css' ), array(), Main::PACKAGE_VERSION . '-' . gmdate( 'oW' ) );
+		}
+
+		wp_enqueue_script(
```

Comment:
TODO: Check if `accessible-focus.js` is used anywhere else and move it to this package if not. This will currently not work.

Reply:
Actually I've confirmed this will still work because `Assets::get_file_url_for_environment` [will set](https://github.com/Automattic/jetpack/blob/0e1ac58f5d1e61d5c74ef236a2649369d0679373/projects/packages/assets/src/class-assets.php#L155) the `$plugin_path` to `Jetpack_Constants::get_constant( 'JETPACK__PLUGIN_FILE' )` if the `$package_path` argument is empty.

As a reminder, this package will be only used by Simple and WoA sites, where the `JETPACK__PLUGIN_FILE` constant should be always defined.

Added a unit test too via https://github.com/Automattic/jetpack/pull/37342/commits/0ecda4fa9e2382ac379ad70a11a276dbbaec44d2